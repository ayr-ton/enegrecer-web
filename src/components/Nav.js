import React from 'react';

import Botao from './Botao';
import { withHistoryPropTypes } from './sharedPropTypes';


const Nav = props => (
  <nav className="en-nav">
    <a className="en-extra-margin-right" href="/sobre">Sobre</a>
    <Botao
      outline
      primary
      onClick={
        () => props.history.push('/painel/login') // eslint-disable-line react/prop-types
      }
    >Registrar / Entrar</Botao>
  </nav>
);

Nav.propTypes = withHistoryPropTypes();

export default Nav;
