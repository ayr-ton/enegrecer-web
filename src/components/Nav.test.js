import React from 'react';
import { shallow } from 'enzyme';

import Nav from './Nav';

describe('Nav', () => {
  const wrapper = shallow(<Nav />);

  it('tem a classe en-nav', () => {
    expect(wrapper.hasClass('en-nav')).toBeTruthy();
  });

  it('tem o link de sobre', () => {
    expect(wrapper.find('[href="/sobre"]').exists()).toBeTruthy();
  });

  it('tem o link de registrar/entrar', () => {
    expect(wrapper.find('[href="/entrar"]').exists()).toBeTruthy();
  });
});
