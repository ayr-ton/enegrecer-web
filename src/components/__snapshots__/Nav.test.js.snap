// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav possui a mesma estrutura 1`] = `
<nav
  className="en-nav"
>
  <a
    className="en-extra-margin-right"
    href="/sobre"
    name="sobre"
  >
    Sobre
  </a>
  <Botao
    colored={false}
    name="Registrar / Entrar"
    onClick={[Function]}
    outline={true}
    primary={true}
  >
    Registrar / Entrar
  </Botao>
</nav>
`;
